### PR TITLE
Migrate to meson

### PR DESCRIPTION
**Pull request by lvitals**
#82

### CODE REVIEW

**Comment by lvitals** [comment]
Done!

 [x] ~~Can you add a LINGUAS file to /po~~
 [x] ~~Can you add a POTFILES file to /po~~

File: meson.build [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+)
+
+i18n = import('i18n')
+
+add_project_arguments(
+    '-DGETTEXT_PACKAGE="@0@"'.format(meson.project_name()),
+    language: 'c'
+)
+
+add_project_arguments(
```
Comment (tintou) [inline meson.build]: This shouldn't be there

File: lib/synapse-core/meson.build [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    configuration: config_data
+)
+
+synapse_core_dir = include_directories('.')
+    
+synapse_core = static_library('synapse-core',
+                        synapse_core_sources,
+                        config_file,
+                        include_directories : synapse_core_dir,
+                        dependencies :  [
```
Comment (tintou) [inline lib/synapse-core/meson.build]: create a synapse_core_deps = […] to reuse it afterward

File: meson.build [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    ['--vapidir', join_paths(meson.current_source_dir(), 'vapi')],
+    language: 'vala'
+)
+
```
Comment (tintou) [inline meson.build]: by convention we all the dependencies here so we can control them in one place only
(things like `glib_dep = dependency('glib-2.0')`)

File: lib/synapse-core/meson.build [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+                            dependency('glib-2.0'),
+                            dependency('gio-unix-2.0'),
+                            dependency('json-glib-1.0'),
+                            dependency('gee-0.8'),
+                            dependency('gtk+-3.0'),
+                            dependency('zeitgeist-2.0')
+                        ],
+                        c_args : '-DGMENU_I_KNOW_THIS_IS_UNSTABLE',
+                        install : true)
```
Comment (tintou) [inline lib/synapse-core/meson.build]: Then do
```meson
synapse_core_dep = declare_dependency(
    link_with: synapse_core,
    include_directories: include_directories('.'),
    dependencies: synapse_core_deps,
)
```

File: lib/synapse-plugins/meson.build [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+                        dependencies :  [
+                            dependency('glib-2.0'),
+                            dependency('gio-unix-2.0'),
+                            dependency('json-glib-1.0'),
+                            dependency('gee-0.8'),
+                            dependency('gtk+-3.0'),
+                            dependency('switchboard-2.0')
+                        ],
+                        c_args : '-DGMENU_I_KNOW_THIS_IS_UNSTABLE',
+                        install : true)
```
Comment (tintou) [inline lib/synapse-plugins/meson.build]: same here, create a `synapse_plugins_dep = declare_dependency(…)`

File: lib/synapse-plugins/meson.build [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    'system-managment.vala',
+    'link-plugin.vala',
+    'appcenter-plugin.vala'
+    ]
+
+synapse_plugins_dir = include_directories('.')
+
+synapse_plugins = static_library('synapse-plugins',
+                        synapse_plugins_sources,
+                        link_with : synapse_core,
```
Comment (tintou) [inline lib/synapse-plugins/meson.build]: no need for link_with and include_directories of synapse_core, just add synapse_core_dep in the `dependencies` list

File: lib/synapse-plugins/meson.build [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+                        include_directories : [synapse_core_dir, synapse_plugins_dir],
+                        dependencies :  [
```
Comment (tintou) [inline lib/synapse-plugins/meson.build]: same as above, create a `synapse_plugins_deps` dependency array

File: src/meson.build [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+]
+
+shared_module(
+    meson.project_name(),
+    files,
+    dependencies: dependencies,
+    vala_args: [
+        meson.source_root() + '/vapi/config.vapi'
+    ],
+    link_with: [synapse_core, synapse_plugins],
```
Comment (tintou) [inline src/meson.build]: no link_with, no include_directories(), only dependencies of synapse_core_dep and synapse_plugins_dep

File: src/meson.build [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+    unity_dep,
+    wingpanel_dep
+]
+
+shared_module(
+    meson.project_name(),
+    files,
+    dependencies: dependencies,
+    vala_args: [
+        meson.source_root() + '/vapi/config.vapi'
```
Comment (tintou) [inline src/meson.build]: don't use vala_args directly here:
```meson
config_vapi = meson.get_compiler('vala').find_library('config', dirs: join_paths(meson.source_root(), 'vapi'))
```
then add config_vapi in the dependencies array

**Comment by tintou** [comment]
Thanks for your contribution, there are some things to change to align it with the elementary standards and make it more easily maintainable :100:

**Comment by lvitals** [comment]
Many thanks for the guidance, changes made.

File: src/meson.build (hunk 8 above)
Comment (lvitals) [reply src/meson.build]: now connected in the library, I can get you out of here.

**Comment by lvitals** [comment]
It was missing compiling the "data" folder. I believe that everything is now in its place.

**Pull request merged**